docker/stacks/local/mysql/docker-compose.yml

```diff
@@ -0,0 +1,102 @@
+version: "3.3"
+
+services:
+
+  mysql:
+    image: mariadb:10.6
+    restart: unless-stopped
+    environment:
+      - MYSQL_ROOT_PASSWORD=qHZDZ26fPqqCfKZtWkQ9
+      - MYSQL_DATABASE=pda
+      - MYSQL_USER=pda
+      - MYSQL_PASSWORD=qHZDZ26fPqqCfKZtWkQ9
+
+  phpmyadmin:
+    image: phpmyadmin/phpmyadmin:5.1
+    depends_on: ['mysql']
+    restart: unless-stopped
+    environment:
+      - PMA_HOST=mysql
+      - PMA_PORT=3306
+    ports:
+      - 8082:80
+
+  powerdns-admin:
+    image: ngoduykhanh/powerdns-admin:latest
+    depends_on: [ 'mysql' ]
+    restart: unless-stopped
+    environment:
+      - PDA_BIND_ADDRESS=0.0.0.0
+      - PDA_BIND_PORT=80
+      - PDA_DEBUG=true
+      - PDA_LOAD_ENV_FILES=true
+      - PDA_AUTOINIT_ENABLED=true
+      - PDA_AUTOINIT_FORCE=false
+      - PDA_CHECK_MYSQL_ENABLED=true
+      - PDA_CHECK_MYSQL_FAIL_DELAY=2
+      - PDA_CHECK_MYSQL_SUCCESS_DELAY=0
+      - PDA_CHECK_MYSQL_ATTEMPTS=30
+      - PDA_CHECK_API_ENABLED=true
+      - PDA_CHECK_API_FAIL_DELAY=2
+      - PDA_CHECK_API_SUCCESS_DELAY=0
+      - PDA_CHECK_API_ATTEMPTS=15
+      - PDA_CHECK_PYTEST_ENABLED=false
+      - PDA_GUNICORN_ENABLED=false
+      - PDA_GUNICORN_TIMEOUT=2
+      - PDA_GUNICORN_WORKERS=2
+      - PDA_GUNICORN_LOGLEVEL=debug
+      - PDA_PDNS_API_KEY=scjfNRN4Ch2DBuhk24vr
+      - PDA_PDNS_VERSION=4.5.2
+      - PDA_PDNS_PROTO=http
+      - PDA_PDNS_HOST=pdns
+      - PDA_PDNS_PORT=8081
+      - PDA_ADMIN_USER_USERNAME=nocadmin
+      - PDA_ADMIN_USER_PASSWORD=clink henchmen mete coracle uphold
+      - PDAC_SALT=ntR8BmMQb7MKTJ5YrgQwZr3pxU3uLG
+      - PDAC_SECRET_KEY=eJ92QLuennVStHTj5FVwxz7m4r3Ng3
+      - PDAC_SIGNUP_ENABLED=true
+      - PDAC_HSTS_ENABLED=false
+      - PDAC_OFFLINE_MODE=false
+      - PDAC_FILESYSTEM_SESSIONS_ENABLED=false
+      - PDAC_SQLA_DB_HOST=mysql
+      - PDAC_SQLA_DB_NAME=pda
+      - PDAC_SQLA_DB_USER=root
+      - PDAC_SQLA_DB_PASSWORD=qHZDZ26fPqqCfKZtWkQ9
+      - PDAC_SQLA_DB_PORT=3306
+      - PDAC_SAML_ENABLED=false
+      - SQLALCHEMY_TRACK_MODIFICATIONS=true
+      - FLASK_ENV=development
+    ports:
+      - "8080:80"
+    volumes:
+      - /PATH/TO/YOUR/GIT/PROJECT/ROOT:/srv/app
```

Comment:
Is it possile to change this to a standard location by using something like $(pwd) ? This way we do not need to edit this file which is handled by git ? or is it intended to be copied outside the repository and edited ?
Solution #2 has two drawbacks: if it is updated in the repo you have to copy and customize again or your docker-compose up could be stale or broken, and two is the ability to spin up a test instance just after pulling.

I'm not completely familiar with docker volume mechanisms. Does this implementation allows us to change code from the host and restart the container to load the changes ?

Reply:
@jbe-dw I will look into that and make it happen if I can. The intent of the current local compose files is to support deploying with both Docker Compose and allowing for direct copy and paste for those using Portainer. One does not have to use the volume mapping on the local, it is simply there as an example of how to most effectively use the image. This would indeed allow for easier reloading of local code changes by simply restarting the container. When configured for the development environment, Flask will run in debug mode which will also auto reload some code base changes.

I also made the /srv/app a volume so that once created, it would be maintained across container recreations. This allows for a number of things to happen here depending on how you actually deploy the image.

Reply:
@jbe-dw Looks like the best option we have here is to use a simple environment variable to control the host path association. I will implement that and update documentation accordingly. Thanks

Reply:
@jbe-dw This change has been implemented.

Reply:
Thank you